icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java

```diff
@@ -2217,6 +2217,9 @@ private String adjustFieldTypes(PatternWithMatcher patternWithMatcher, DateTimeM
                             && (type != YEAR || reqFieldChar=='Y'))
                             ? reqFieldChar
                             : fieldBuilder.charAt(0);
+                    if (c == 'E' && adjFieldLen < 3) {
```

Comment:
ditto

Reply:
Ditto

Reply:
could you add the comments in the code, not just the test code. thanks